File: src/main/java/javabot/operations/time/TimezoneParser.java
package javabot.operations.time;

public class TimezoneParser {

    public Timezone getTimezone( Country country, String province ) {
        Timezone timezone;

        if ( country.isMultiZones( ) ) {
            Timezones timezones = country.getMultiZones( );
            String timezoneId = timezones.get( province );

            if ( timezoneId != null ) {
                timezone = new Timezone( timezoneId, province );
            } else {
                timezone = new Timezone( timezones.getCapital( ), country.getName( ) );
            }
        } else {
            timezone = new Timezone( country.getTimezone( ), country.getName( ) );
        }
        return timezone;
    }

    public class Timezone {
        String timezone;
        String region;

        public Timezone( String timezone, String region ) {
            this.timezone = timezone;
            this.region = region;
        }

        public String getTimezone( ) {
            return timezone;
        }

        public String getRegion( ) {
            return region;
        }
    }
}

File: src/main/kotlin/javabot/operations/time/TimezoneParser.kt
package javabot.operations.time

public class TimezoneParser {

    public fun getTimezone(country: Country, province: String): Timezone {
        val timezone: Timezone

        if (country.isMultiZones) {
            val timezones = country.multiZones
            val timezoneId = timezones.get(province)

            if (timezoneId != null) {
                timezone = Timezone(timezoneId, province)
            } else {
                timezone = Timezone(timezones.capital, country.name)
            }
        } else {
            timezone = Timezone(country.timezone, country.name)
        }
        return timezone
    }

    public inner class Timezone(timezone: String, region: String) {
        public var timezone: String
            set
        public var region: String
            set

        init {
            this.timezone = timezone
            this.region = region
        }
    }
}
